src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/group/GroupVerifyCommand.java

```diff
@@ -0,0 +1,132 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.command.group;
+
+import static java.util.stream.Collectors.toList;
+
+import org.openkilda.floodlight.command.SpeakerCommandProcessor;
+import org.openkilda.floodlight.converter.OfFlowStatsMapper;
+import org.openkilda.floodlight.error.SwitchIncorrectMirrorGroupException;
+import org.openkilda.floodlight.error.SwitchMissingGroupException;
+import org.openkilda.floodlight.error.UnsupportedSwitchOperationException;
+import org.openkilda.floodlight.utils.CompletableFutureAdapter;
+import org.openkilda.messaging.MessageContext;
+import org.openkilda.messaging.info.rule.GroupBucket;
+import org.openkilda.messaging.info.rule.GroupEntry;
+import org.openkilda.model.GroupId;
+import org.openkilda.model.MirrorConfig;
+import org.openkilda.model.MirrorConfig.MirrorConfigData;
+import org.openkilda.model.SwitchId;
+
+import org.projectfloodlight.openflow.protocol.OFGroupDescStatsEntry;
+import org.projectfloodlight.openflow.protocol.OFGroupDescStatsReply;
+import org.projectfloodlight.openflow.protocol.OFGroupDescStatsRequest;
+import org.projectfloodlight.openflow.protocol.match.MatchField;
+import org.projectfloodlight.openflow.types.DatapathId;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+public class GroupVerifyCommand extends AbstractGroupInstall<GroupVerifyReport> {
+    public GroupVerifyCommand(
+            MessageContext messageContext, SwitchId switchId, MirrorConfig mirrorConfig) {
+        super(messageContext, switchId, mirrorConfig);
+    }
+
+    @Override
+    protected CompletableFuture<GroupVerifyReport> makeExecutePlan(SpeakerCommandProcessor commandProcessor)
+            throws UnsupportedSwitchOperationException {
+        ensureSwitchSupportGroups();
+
+        return new CompletableFutureAdapter<>(
+                messageContext, getSw().writeStatsRequest(makeGroupReadCommand()))
+                .thenApply(this::handleGroupStats)
+                .thenApply(this::makeSuccessReport);
+    }
+
+    @Override
+    protected GroupVerifyReport makeReport(Exception error) {
+        return new GroupVerifyReport(this, error);
+    }
+
+    private GroupVerifyReport makeSuccessReport(MirrorConfig mirrorConfig) {
+        return new GroupVerifyReport(this, mirrorConfig);
+    }
+
+    private OFGroupDescStatsRequest makeGroupReadCommand() {
+        return getSw().getOFFactory().buildGroupDescStatsRequest().build();
```

Comment:
Can we request only specific group id?

Reply:
No, we can not...
`org.projectfloodlight.openflow.protocol.ver13.OFGroupDescStatsRequestVer13.Builder#setGroup`
"Property group not supported in version 1.3"